core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -237,21 +237,24 @@ private fun <ReqT : ResourceRequirement> detectRequirementConflicts(
 }
 
 enum class EventType { BEGIN, END }
-class Event (eventType: EventType, time : Double){
-
-    val time: Double = time
-    val eventType : EventType = eventType
+class Event (val eventType: EventType, val time : Double) : Comparable<Event>
+{
+    override fun compareTo(other: Event): Int {
+        return when (this.eventType) {
+            other.eventType -> 0
+            EventType.BEGIN -> -1
+            EventType.END -> 1
+        }
```

Comment:
you need to compare by time first, otherwise things won't be sorted properly
```suggestion
        val timeDelta = this.time.compareTo(other.time)
        if (timeDelta != 0)
            return timeDelta
        return when (this.eventType) {
            other.eventType -> 0
            EventType.BEGIN -> -1
            EventType.END -> 1
        }
```